pkgs/development/python-modules/array-api-compat/default.nix

```
@@ -0,0 +1,63 @@
+{
+  lib,
+  buildPythonPackage,
+  fetchFromGitHub,
+  pytestCheckHook,
+  setuptools,
+  numpy,
+  jaxlib,
+  jax,
+  torch,
+  dask,
+  sparse,
+  array-api-strict,
+  config,
+  cudaSupport ? config.cudaSupport,
+  cupy,
+  nix-update-script,
+}:
+
+buildPythonPackage rec {
+  pname = "array-api-compat";
+  version = "1.8";
+  pyproject = true;
+
+  src = fetchFromGitHub {
+    owner = "data-apis";
+    repo = "array-api-compat";
+    rev = "refs/tags/${version}";
+    hash = "sha256-DZs51yWgeMX7lmzR6jily0S3MRD4AVlk7BP8aU99Zp8=";
+  };
+
+  build-system = [ setuptools ];
+
+  nativeCheckInputs = [
+    pytestCheckHook
+    numpy
+    jaxlib
+    jax
+    torch
+    dask
+    sparse
+    array-api-strict
+  ] ++ lib.optionals cudaSupport [ cupy ];
+
+  pythonImportsCheck = [ "array_api_compat" ];
+
+  # CUDA (used via cupy) is not available in the testing sandbox
+  checkPhase = ''
+    runHook preCheck
+    python -m pytest -k 'not cupy'
+    runHook postCheck
+  '';
```

```suggestion
  pytestFlagsArray = [
    "-k"
    "'not cupy'"
  ];
```
see https://github.com/NixOS/nixpkgs/blob/master/doc/languages-frameworks/python.section.md#using-pytestcheckhook-using-pytestcheckhook